From foo@baz Tue Apr  9 12:12:43 2002
To: Greg KH <greg@kroah.com>
Date: 02 Mar 2005 15:02:43 -08:00
From: ben-linux@fluff.org
Subject: [PATCH] I2C: S3C2410 missing I2C_CLASS_HWMON

None of the standard sensor drivers currently recognise the s3c24xx
I2C controller as it does not have I2C_CLASS_HWMON set in the
adapter class field.

The attached patch initialises the adapter class to I2C_CLASS_HWMON

Signed-off-by: Ben Dooks <ben-linux@fluff.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


diff -Nru a/drivers/i2c/busses/i2c-s3c2410.c b/drivers/i2c/busses/i2c-s3c2410.c
--- a/drivers/i2c/busses/i2c-s3c2410.c	2005-03-02 16:32:02 -08:00
+++ b/drivers/i2c/busses/i2c-s3c2410.c	2005-03-02 16:32:02 -08:00
@@ -569,6 +569,7 @@
 		.name			= "s3c2410-i2c",
 		.algo			= &s3c24xx_i2c_algorithm,
 		.retries		= 2,
+		.class			= I2C_CLASS_HWMON,
 	},
 };
 
